### PR TITLE
Preserve unicode when slugifying by default

### PR DESCRIPTION
**Pull request by rtpg**
Thanks to @hwshim810 (for original implementation) and @lucemia 
(for ping + pointing out the perf weirdness) from #572 for motivating this.
 
 This change is motivated by the many many people using non-latin
languages who experience the (rather baffling) behavior of outright
stripping characters when generating slugs from stuff that doesn't
fit into ASCII.

 We went through loads of pain as a programming community to get to
nicely supporting unicode everywhere, it's time to take advantage of
that fact and just let people have stuff appear in their native
language as much as possible.

### CODE REVIEW

**Comment by rtpg** [comment]
@lucemia would you be able to review thise changeset? given this is a pretty big change (in few lines of code of course) I do want to have at least one review from someone else

**Comment by lucemia** [comment]
> @lucemia would you be able to review thise changeset? given this is a pretty big change (in few lines of code of course) I do want to have at least one review from someone else

Would love to, but I am not reviewer of this repo...

**Comment by rtpg** [comment]
Two options:
- you can just look at the code and make a comment. I want somebody to look over it but I am totally comfortable hitting the merge button if it looks good myself. You don't need to be a "real" reviewer.
- you can join the jazzband org (see [here](https://jazzband.co/) for more details) and do the review "for real"

File: CHANGELOG.rst
Comment (lucemia) [inline CHANGELOG.rst:24]: I will be very helpful to add description `TAGGIT_STRIP_UNICODE__WHEN_SLUGIFYING` in the document as well.
e.g. https://github.com/jazzband/django-taggit/blob/master/docs/getting_started.rst .  in the `Note` section.

Good to also mention that there is another option: install `unidecode` (translate all non ascii character to ascii first). These information is only address in [changelog](https://django-taggit.readthedocs.io/en/latest/changelog.html#id43)

https://github.com/jazzband/django-taggit/blob/2626e235a65f510e1282bccc1a2f7568506642a3/taggit/models.py#L10-L16


```
>>> unidecode.unidecode('あい うえお')
'ai ueo'
```

**Comment by lucemia** [comment]
> Two options:
> 
> * you can just look at the code and make a comment. I want somebody to look over it but I am totally comfortable hitting the merge button if it looks good myself. You don't need to be a "real" reviewer.
> * you can join the jazzband org (see [here](https://jazzband.co/) for more details) and do the review "for real"

I joined Jazzband!

**Review by lucemia — APPROVED** [review]

**Comment by rtpg** [comment]
https://github.com/jazzband/django-taggit/pull/798 this PR seems to indicate another important change that might be necessary here: `allow_unicode=True` for `SlugField`. Will look at this issue to see if something should be done here

**Comment by rtpg** [comment]
Alright I made sure that `SlugField` would accept unicode, and improved the documentation. @lucemia if you could look over this one last time and we can merge it and then I'll prepare a release

**Comment by lucemia** [comment]
I checked the PR, it looks good to me!
Thanks your @rtpg  nice work!

**Comment by jsonn** [comment]
Are you aware that the test case here is broken with Python 3.8?

**Comment by rtpg** [comment]
@jsonn Python 3.8 is within the test suite and is passing, do you have a stack trace showing failure that can be tracked down here?

**Comment by jsonn** [comment]
```
Traceback (most recent call last):
  File "/tmp/scratch/www/py-django-taggit/work/django-taggit-3.1.0/tests/test_models.py", line 27, in test_old_slugs
    self.assertEqual([tag.slug for tag in sample_obj.tags.all()], [""])
AssertionError: Lists differ: ['ai-ueo'] != ['']

First differing element 0:
'ai-ueo'
''

- ['ai-ueo']
+ ['']
```

**Pull request merged**
